public_html/src/Demos/ConditionalDemo/ConditionalDemo.js

```diff
@@ -0,0 +1,599 @@
+            return true;
+        }
+        return false;
+    }
+    
+    _setupActionRenderable() {
+        var sprite = this.mTurnSystem.getCurrentUser().getMarker();
+        var position = this.mParameters.mPositionDict[this.mCurAction];
+        var width = this.mParameters.mCurActionRenWidth;
+        
+        var ren = new SpriteRenderable(sprite);
+        ren.setColor([1, 1, 1, 0]);
+        ren.getXform().setPosition(position[0], position[1]);
+        ren.getXform().setSize(width, width);
+        ren.setElementPixelPositions(0, 512, 0, 512);
+        
+        this.mActionsRens.push(ren);
+    }
+    
+    _modifyActionArr() {
+        var row = Math.floor((this.mCurAction - 1)/3);
+        var col = (this.mCurAction - 1)%3;
+        this.mActionsArr[row][col] = this.mTurnSystem.getCurrentUser().getActionType();
+    }
+    
+    _setupCurActionRenderable() {
+        if (this.mCurActionRen === null) {
+            var boxWidth = this.mParameters.mCurActionRenWidth;
+            // Create the renderable
+            this.mCurActionRen = new Renderable();
+            this.mCurActionRen.setColor([1,0,0,1]);
+            this.mCurActionRen.getXform().setSize(boxWidth, boxWidth);
+        }
+        
+        var position = this.mParameters.mPositionDict[this.mCurAction];
+        // Translate the position based on action
+        this.mCurActionRen.getXform().setPosition(position[0],position[1]);
+    }
+    
+    // Check user's input
+    _userInput() {
+        // End turn
+        if (gEngine.Input.isKeyClicked(gEngine.Input.keys.E)) {
+            if (this.mTurnSystem.getCurrentUser().mIsMadeMove === true) {
+                this.mTurnSystem.getCurrentUser().isMetCondition = true;
+                
+                // setup the renderable at action position
+                this._setupActionRenderable();
+                
+                // modify the action array to store the board
+                this._modifyActionArr();
+                
+                // reset current action
+                this.mCurAction = null;
+                this.mCurActionRen = null;
+                this.mTurnSystem.getCurrentUser().mIsMadeMove = false;
+            }
+        }
+        
+        if (gEngine.Input.isButtonClicked(gEngine.Input.mouseButton.Left)) {
+            var action = this._playerAction();
+            //console.log(`action = ${action}`);
+            if (action !== actions.NONE && this._isValidAction(action)) {
+                this.mCurAction = action;
+                this._setupCurActionRenderable();
+                this.mTurnSystem.getCurrentUser().mIsMadeMove = true;
+            }
+        }
+    }
+    
+    // Update the text in the CurrentPlayer font renderable based on current player
+    _updateCurrentPlayer() {
+        var curPlayer = this.mTurnSystem.getCurrentUser();
+        this.mCurrentPlayer.setText("Current Player: Player " + (curPlayer.getIndex() + 1));
```

Comment:
```suggestion
        this.mCurrentPlayer.setText(`Current Player: Player ${curPlayer.getIndex() + 1}`);
```
This should give the same results. It's just a bit cleaner